wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java

```diff
@@ -338,13 +338,12 @@ public FactoryDto resolveFactory(
     requiredNotNull(parameters, "Factory build parameters");
 
     // search matching resolver and create factory from matching resolver
-    FactoryDto resolvedFactory =
-        factoryParametersResolverHolder
-            .getFactoryParametersResolver(parameters)
-            .createFactory(parameters);
-    if (resolvedFactory == null) {
+    FactoryParametersResolver resolver =
+        factoryParametersResolverHolder.getFactoryParametersResolver(parameters);
+    if (resolver == null) {
       throw new BadRequestException(FACTORY_NOT_RESOLVABLE);
     }
+    FactoryDto resolvedFactory = resolver.createFactory(parameters);
```

Comment:
you've removed null check for this. Looking at https://github.com/eclipse/che/blob/master/wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/DefaultFactoryParameterResolver.java#L70 I guess we should keep check for null.

Reply:
fixed